### PR TITLE
[PAY-891] Handle remix and availability settings on mobile

### PR DESCRIPTION
**Pull request by sddioulde**
### Description

Handle remix and availability settings on mobile.
Also fix web remix settings.

### Dragons

track upload flow

### How Has This Been Tested?

local ios app and local web app vs stage

### How will this change be monitored?

n/a

### Feature Flags ###

existing premium content feature flags

### CODE REVIEW

File: packages/common/src/services/solana-client/SolanaClient.ts [hunk 1]
```diff
@@ -175,7 +175,8 @@ export class SolanaClient {
         this.connection,
         programAddress
       )
-      const result = (await (await fetch(metadata.data.uri)).json()) ?? {}
+      const response = await fetch(metadata.data.uri.replaceAll('\x00', ''))
```
Comment (sddioulde) [inline packages/common/src/services/solana-client/SolanaClient.ts:178]: was failing all of a sudden because of weird trailing chars

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-pc-settings-mobile

File: packages/mobile/src/screens/edit-track-screen/screens/RemixSettingsScreen.tsx [hunk 2]
```diff
@@ -90,6 +131,17 @@ export const RemixSettingsScreen = () => {
...
   const isInvalidParentTrack = parentTrackStatus === Status.ERROR
 
+  useEffect(() => {
+    if (isPremium) {
+      setRemixOf(null)
+      setRemixesVisible(false)
+    } else {
+      setRemixesVisible(true)
+    }
+    // adding the useField setters cause infinite rendering
```
Comment (rickyrombo) [inline packages/mobile/src/screens/edit-track-screen/screens/RemixSettingsScreen.tsx:141]: this sounds like a bug

**Review by rickyrombo — APPROVED** [review]

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-pc-settings-mobile

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-pc-settings-mobile

**Pull request merged**